src/atoms/referencesState.ts

```diff
@@ -0,0 +1,31 @@
+import { atom } from 'jotai';
+
+import { ReferenceItem } from '../types/ReferenceItem';
+
+const pdfIngestionProcessAtom = atom<string[]>([]); // ids of pending processes for now, this could be a record with progress percentage in the future
+
+export const isPdfIngestionRunningAtom = atom((get) => get(pdfIngestionProcessAtom).length > 0);
+
+export const addPdfIngestionProcessAtom = atom(null, (_get, set, value: string) => {
+  set(pdfIngestionProcessAtom, (pendingProcesses) => [...pendingProcesses, value]);
+});
```

Comment:
Given that this logic is so simple (push to array), shouldn't one use the `pdfIngestionProcessAtom` instead?

Reply:
Yes, the logic is simple for now, but I had in mind that I could get more complex in the future, like if we want to store a progress percentage along with the progress. This atom is the atom allowing a component to say "An ingestion progress has started" without knowing anything about the underlying representation we have chosen to store processes.
I agree in this case this may be overcomplicating and we can actually just expose the `pdfIngestionProcessAtom`